File: PokePlannerWeb.Data/Types/TypeSet.cs
namespace PokePlannerWeb.Data.Types
{
    /// <summary>
    /// Class mapping types to whether they're present in a version group.
    /// </summary>
    public class TypeSet
    {
        /// <summary>
        /// Gets or sets the version group ID.
        /// </summary>
        public int VersionGroupId { get; set; }

        /// <summary>
        /// Gets or sets the types.
        /// </summary>
        public string[] Types { get; set; }

        /// <summary>
        /// Gets or sets whether the types are present.
        /// </summary>
        public bool[] TypesArePresent { get; set; }
    }
}

File: PokePlannerWeb/ClientApp/src/models/TypeSet.ts
import { IHasVersionGroup } from "../components/CommonMembers"

/**
 * Class mapping types to whether they're present in a version group.
 */
export class TypeSet implements IHasVersionGroup {
    /**
     * The version group ID.
     */
    versionGroupId: number | undefined

    /**
     * The types.
     */
    typeIds: number[]

    /**
     * Whether the types are present.
     */
    typesArePresent: any[]

    /**
     * Default constructor.
     * @param versionGroupId The version group ID.
     * @param typeIds The IDs of the types.
     * @param typesArePresent Whether the types are present.
     */
    constructor(versionGroupId: number | undefined, typeIds: number[], typesArePresent: any[]) {
        this.versionGroupId = versionGroupId
        this.typeIds = typeIds
        this.typesArePresent = typesArePresent
    }
}